R/preprocess.R

```diff
@@ -70,34 +73,15 @@ preprocess_data <- function(dataset, outcome_colname,
   check_outcome_column(dataset, outcome_colname, check_values = FALSE)
   check_remove_var(remove_var)
   pbtick(progbar)
-  dataset[[outcome_colname]] <- replace_spaces(dataset[[outcome_colname]])
-  dataset <- rm_missing_outcome(dataset, outcome_colname)
-  split_dat <- split_outcome_features(dataset, outcome_colname)
-
-  features <- split_dat$features
-  removed_feats <- character(0)
-  if (to_numeric) {
-    feats <- change_to_num(features) %>%
-      remove_singleton_columns(threshold = prefilter_threshold)
-    removed_feats <- feats$removed_feats
-    features <- feats$dat
-  }
-  pbtick(progbar)
-
-  nv_feats <- process_novar_feats(features, progbar = progbar)
-  pbtick(progbar)
-  split_feats <- process_cat_feats(nv_feats$var_feats, progbar = progbar)
-  pbtick(progbar)
-  cont_feats <- process_cont_feats(split_feats$cont_feats, method)
-  pbtick(progbar)
-
+  
+  processed_data <- prep_data(dataset, outcome_colname, prefilter_threshold, method, impute_in_preprocessing, to_numeric)
```

Comment:
Why do you think moving this code to a separate `prep_data` function is necessary?

Reply:
Oh, I see that you also called it in `run_ml` below. I think we don't need `prep_data', because in `run_ml` it should be possible to _only_ impute missing values after the split and _not_ do any other preprocessing steps there. (see other comment below)

Reply:
I see what you are saying. We removed the prep_data function and instead added the step to compute n_missing to the impute function.